### ISSUE TITLE
8.7.0 UI docker image error

### ISSUE THREAD

**Issue by CodingOX** [apache/skywalking#7401]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [x] Bug

___
### Bug
- Which version of SkyWalking, OS, and JRE?
Skywalking:8.7.0
OS:Docker

- What happened?
After deployment through docker, a blank page will appear when accessing through the browser, and the errors reported on the console are as follows
```shell
2021-08-03 03:29:17.487 ERROR 1 --- [or-http-epoll-3] a.w.r.e.AbstractErrorWebExceptionHandler : [1816cafd-1]  500 Server Error for HTTP POST "/graphql"

io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /127.0.0.1:12800
        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
        |_ checkpoint ⇢ org.springframework.cloud.gateway.filter.WeightCalculatorWebFilter [DefaultWebFilterChain]
        |_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
        |_ checkpoint ⇢ HTTP POST "/graphql" [ExceptionHandlingWebHandler]
Stack trace:
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
        at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155)
        at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128)
        at io.netty.channel.unix.Socket.finishConnect(Socket.java:278)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)

2021-08-03 03:29:23.184 ERROR 1 --- [or-http-epoll-6] a.w.r.e.AbstractErrorWebExceptionHandler : [d4bc1a05-1]  500 Server Error for HTTP POST "/graphql"

io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /127.0.0.1:12800
        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
        |_ checkpoint ⇢ org.springframework.cloud.gateway.filter.WeightCalculatorWebFilter [DefaultWebFilterChain]
        |_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
        |_ checkpoint ⇢ HTTP POST "/graphql" [ExceptionHandlingWebHandler]
Stack trace:
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
        at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155)
        at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128)
        at io.netty.channel.unix.Socket.finishConnect(Socket.java:278)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)

2021-08-03 03:29:43.656 ERROR 1 --- [or-http-epoll-7] a.w.r.e.AbstractErrorWebExceptionHandler : [816cd19d-2]  500 Server Error for HTTP POST "/graphql"

io.netty.channel.AbstractChannel$AnnotatedConnectException: finishConnect(..) failed: Connection refused: /127.0.0.1:12800
        Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
        |_ checkpoint ⇢ org.springframework.cloud.gateway.filter.WeightCalculatorWebFilter [DefaultWebFilterChain]
        |_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
        |_ checkpoint ⇢ HTTP POST "/graphql" [ExceptionHandlingWebHandler]
Stack trace:
Caused by: java.net.ConnectException: finishConnect(..) failed: Connection refused
        at io.netty.channel.unix.Errors.newConnectException0(Errors.java:155)
        at io.netty.channel.unix.Errors.handleConnectErrno(Errors.java:128)
        at io.netty.channel.unix.Socket.finishConnect(Socket.java:278)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.doFinishConnect(AbstractEpollChannel.java:710)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.finishConnect(AbstractEpollChannel.java:687)
        at io.netty.channel.epoll.AbstractEpollChannel$AbstractEpollUnsafe.epollOutReady(AbstractEpollChannel.java:567)
        at io.netty.channel.epoll.EpollEventLoop.processReady(EpollEventLoop.java:470)
        at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:378)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
```

The docker deployment script is as follows
```shell
docker pull apache/skywalking-oap-server:8.7.0-es7

docker run --name oap-server --restart always \
 -p 11800:11800 -p 12800:12800 \
 --net=sky-fly \
 -e SW_STORAGE=elasticsearch7 \
 -e SW_STORAGE_ES_CLUSTER_NODES=elasticsearch:9200 \
 -d apache/skywalking-oap-server:8.7.0-es7

docker pull apache/skywalking-ui:8.7.0

docker run --name oap-ui --restart always \
 -p 12889:8080 \
 --net=sky-fly \
 -e SW_OAP_ADDRESS=oap-server:12800 \
 -d apache/skywalking-ui:8.7.0
```

With the same deployment method, there is no problem with 8.6.0

___
### Requirement or improvement
Bug or my fault?

**Comment by wu-sheng** [comment]
This is your deployment, you should check on your own. 500 means backend is not available, mostly, you network setting is wrong.

**Comment by CodingOX** [comment]
In version 8.7, this problem can be 100% replicated in multiple test environments, but  8.6 work fine. So I don't think my network wrong.

**Comment by wu-sheng** [comment]
It says `Connection refused`, there is nothing we can do.

**Comment by CodingOX** [comment]
`there is nothing we can do.` This is a rather irresponsible reply. If there is no problem with my deployment process, as the core developer of skywaking, Can you take a little time to verify this error?Of course, you can say that open source authors have no obligation to waste their time reproducing other people's problems.I've always been uncomfortable with what  you rude shut down issue, maybe your sign is Virgo.Skywalking is really a very good product.
Let's get to the point

---
The reason for the problem is that the UI images of 8.7 in the current dockerhub is wrong, `docker-entrypoint.sh` content is 

```shell
#!/bin/bash

set -e

export LOGGING_CONFIG="webapp/logback.xml"

[[ ! -z "$SW_OAP_ADDRESS" ]] && export COLLECTOR_RIBBON_LISTOFSERVERS=${SW_OAP_ADDRESS} && echo "COLLECTOR_RIBBON_LISTOFSERVERS=$COLLECTOR_RIBBON_LISTOFSERVERS"
[[ ! -z "$SW_TIMEOUT" ]] && export COLLECTOR_RIBBON_READTIMEOUT=${SW_TIMEOUT} && echo "COLLECTOR_RIBBON_READTIMEOUT=$COLLECTOR_RIBBON_READTIMEOUT"

exec java -jar webapp/skywalking-webapp.jar "$@"
```
but 8.7 changed from zuul to gateway.

The following link is correct, but it is not packaged in the image：https://github.com/apache/skywalking/blob/master/docker/ui/docker-entrypoint.sh
```shell
set -e

export LOGGING_CONFIG="webapp/logback.xml"

if [[ ! -z "$SW_OAP_ADDRESS" ]]; then
  address_arr=(${SW_OAP_ADDRESS//,/ })
  for i in "${!address_arr[@]}"
  do
      JAVA_OPTS="${JAVA_OPTS} -Dspring.cloud.discovery.client.simple.instances.oap-service[$i].uri=${address_arr[$i]}"
  done
fi

exec java  ${JAVA_OPTS} -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap -jar webapp/skywalking-webapp.jar "$@"
```

To sum up, I guess there was a problem when packing ui-images.

**Comment by wu-sheng** [comment]
@hanahmily Please check.

**Comment by JaredTan95** [comment]
Yes, we should update the entry-point script in UI docker image.

**Comment by JaredTan95** [comment]
@CodingOX Thank you for your timely feedback.
In addition, we encourage skywalking users to fix and contribute.
Many of the core developers(including me) in skywalking are interested in personal interests and use their spare time to maintain skywalking.

**Comment by JaredTan95** [comment]
@CodingOX Thank you for your timely feedback.
In addition, Skywalker is encouraged to repair and contribute.
Many core developers (including me) because of their personal interests and use their spare time to keep Skywalking.

**Comment by wu-sheng** [comment]
Image updated by @hanahmily . FYI @JaredTan95

**Comment by sundong1982** [comment]
I also encountered this problem with ui: 8.7.0. Replacing it with ui:8.6.0 can solve this problem.

**Comment by wu-sheng** [comment]
> I also encountered this problem with ui: 8.7.0. Replacing it with ui:8.6.0 can solve this problem.

You don't need to. Download the image again, it is fixed on the central docker hub.

**Comment by Zhang21** [comment]
same problem with `apache/skywalking-ui:8.7.0`.

**Comment by Zhang21** [comment]
answer: <https://github.com/apache/skywalking/issues/7416>

**Comment by sundong1982** [comment]
@wu-sheng  It's still old in hub.docker.com

**Comment by wu-sheng** [comment]
I don't know why you pinged me. But this has been updated. Read https://github.com/apache/skywalking/issues/7416#issuecomment-899331509

**Comment by haiyes** [comment]
可以把这个配置调整试试看，本人在8.7版本已经验证通过
SW_OAP_ADDRESS=**oap-server**:12800
SW_OAP_ADDRESS=**http://127.0.0.1**:12800

**Comment by haiyes** [comment]
![image](https://user-images.githubusercontent.com/26479419/131479277-abe22634-a0b6-4369-af44-e1660b04cdce.png)
